minimesos/src/test/java/com/containersol/minimesos/ConsulRegistrationTest.java

```diff
@@ -0,0 +1,55 @@
+package com.containersol.minimesos;
+
+import com.containersol.minimesos.cluster.MesosCluster;
+import com.containersol.minimesos.config.ConsulConfig;
+import com.containersol.minimesos.config.RegistratorConfig;
+import com.containersol.minimesos.docker.DockerContainersUtil;
+import com.containersol.minimesos.marathon.Marathon;
+import com.containersol.minimesos.mesos.*;
+import com.github.dockerjava.api.DockerClient;
+import com.mashape.unirest.http.Unirest;
+import com.mashape.unirest.http.exceptions.UnirestException;
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.junit.After;
+import org.junit.ClassRule;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+
+public class ConsulRegistrationTest {
+
+    protected static final DockerClient dockerClient = DockerClientFactory.build();
+    protected static final ClusterArchitecture CONFIG = new ClusterArchitecture.Builder(dockerClient)
+            .withZooKeeper()
+            .withMaster()
+            .withAgent(zooKeeper -> new MesosAgent(dockerClient, zooKeeper))
+            .withMarathon(zooKeeper -> new Marathon(dockerClient, zooKeeper))
+            .withConsul(new Consul(dockerClient, new ConsulConfig()))
+            .withRegistrator(consul -> new Registrator(dockerClient, consul, new RegistratorConfig()))
+            .build();
+
+    @ClassRule
+    public static final MesosCluster CLUSTER = new MesosCluster(CONFIG);
+
+    @After
+    public void after() {
+        DockerContainersUtil util = new DockerContainersUtil(CONFIG.dockerClient);
+        util.getContainers(false).filterByName(HelloWorldContainer.CONTAINER_NAME_PATTERN).kill().remove();
```

Comment:
Apparently this already works as intended. I copied this code from the `MesosClusterTest` class. 

The filter returns zero containers when there are none, and the chain exits.

Hmm, I've tried changing the file pattern to something that does not exist and even then the hello-world container is removed. Maybe we don't need this `@After` at all? Who removes the containers?